File: supermarket_flutter_app/lib/core/models/category.dart
class Category {
  final int? id;
  final String name;
  final String? description;
  final String? imageUrl;
  final DateTime? createdAt;

  Category({
    this.id,
    required this.name,
    this.description,
    this.imageUrl,
    this.createdAt,
  });

  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      imageUrl: json['imageUrl'],
      createdAt: json['createdAt'] != null ? DateTime.parse(json['createdAt']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'name': name,
      if (description != null) 'description': description,
      if (imageUrl != null) 'imageUrl': imageUrl,
    };
  }
}

File: SupermarketAPI/Models/Category.cs
using System.ComponentModel.DataAnnotations;

namespace SupermarketAPI.Models
{
    public class Category
    {
        public int Id { get; set; }

        [Required]
        [StringLength(128)]
        public string Name { get; set; } = string.Empty;

        [StringLength(1024)]
        public string? ImageUrl { get; set; }
    }
}
